src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -175,6 +192,51 @@ public DRSObject lookupObjectByDrsId(
     }
   }
 
+  public DRSAccessURL getAccessUrlForObjectId(
+      AuthenticatedUserRequest authUser, String objectId, String accessId) {
+    DRSObject drsObject = lookupObjectByDrsId(authUser, objectId, false);
+
+    DrsId drsId = drsIdService.fromObjectId(objectId);
+    Snapshot snapshot = snapshotService.retrieve(UUID.fromString(drsId.getSnapshotId()));
+    GoogleProjectResource projectResource = snapshot.getProjectResource();
+
+    List<DRSAccessMethod> matchingAccessMethods =
+        getAccessMethodsMatchingAccessId(accessId, drsObject);
+
+    if (matchingAccessMethods.size() > 0) {
```

Comment:
Yeah, it'd be great to avoid this awkwardness by just having `getAccessMethodsMatchingAccessId` return an optional.  You could end up with something like:
```
DRSAccessMethod matchingAccessMethods =
        getAccessMethodMatchingAccessId(accessId, drsObject)
           .orElseThrow(new IllegalArgumentException(
               "No matching access ID " + accessId + " was found on object " + objectId));
```